### PR TITLE
Rewrite tests to use `serde_json::json!` instead of raw strings

### PR DESCRIPTION
**Pull request by NerdyPepper**
- [x] event_type.rs
- [x] presence.rs
- [x] ignored_user_list.rs
- [x] push_rules.rs
- [x] to_device.rs
- [x] direct.rs
- [x] dummy.rs
- [x] stripped.rs
- [x] key/verification/start.rs
- [x] key/verification/cancel.rs
- [x] room/pinned_events.rs
- [x] room/name.rs
- [x] room/canonical_alias.rs
- [x] room/member.rs
- [x] room/create.rs
- [x] room/power_levels.rs
- [x] room/message.rs
- [x] room/encrypted.rs
- [x] room/server_acl.rs

### CODE REVIEW

File: src/direct.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ mod tests {
     use std::collections::HashMap;
 
     use ruma_identifiers::{RoomId, UserId};
-    use serde_json::to_string;
+    use serde_json::{from_value, json, to_value};
```
Comment (jplatte) [inline src/direct.rs]: Please only import `json!` and use the other items by their absolute path. Alternatively, we could establish the convention of renaming them to `from_json_value` and `to_json_value` locally. What do you think?

File: src/push_rules.rs [hunk 2]
```diff
@@ -444,37 +444,47 @@ mod tests {
...
 
     #[test]
     fn deserialize_tweak_sound_action() {
+        let json_data = json!({
+            "set_tweak": "sound",
+            "value": "default"
+        });
         assert_eq!(
-            from_str::<Action>(r#"{"set_tweak":"sound","value":"default"}"#).unwrap(),
+            from_value::<Action>(json_data).unwrap(),
```
Comment (jplatte) [inline src/push_rules.rs]: Feel free to remove the redundant generic arguments in this PR:

```suggestion
            from_value(json_data).unwrap(),
```

File: src/direct.rs (hunk 1 above)
Comment (NerdyPepper) [reply src/direct.rs]: I am leaning towards the convention of renaming the imports. It would look a lot cleaner imo.

Comment (jplatte) [reply src/direct.rs]: Okay. Let's do that, then.

File: src/push_rules.rs (hunk 2 above)
Comment (NerdyPepper) [reply src/push_rules.rs]: ah it seems we need that type parameter
 ```
error[E0283]: type annotations needed
   --> src/push_rules.rs:489:13
    |
489 |             from_json_value(json_data).unwrap(),
    |             ^^^^^^^^^^^^^^^--------------------
    |             |
    |             this method call resolves to `T`
    |             cannot infer type for type parameter `T` declared on the function `from_value`
    |             help: consider specifying the type argument in the function call: `from_json_value::<T>`
```

Comment (jplatte) [reply src/push_rules.rs]: Ah, I thought it could be inferred here. Nevermind then.

File: src/direct.rs [hunk 3]
```diff
@@ -25,7 +25,9 @@ mod tests {
     use std::collections::HashMap;
 
     use ruma_identifiers::{RoomId, UserId};
-    use serde_json::{from_value, json, to_value};
+    use serde_json::from_value as from_json_value;
+    use serde_json::json;
+    use serde_json::to_value as to_json_value;
```
Comment (jplatte) [inline src/direct.rs]: You don't need to split up the import to rename items in it.

```suggestion
    use serde_json::{from_value as from_json_value, json, to_value as to_json_value};
```

File: src/to_device.rs [hunk 4]
```diff
@@ -627,23 +629,26 @@ mod tests {
...
+        });
 
         let event = deserialize! {source, AnyToDeviceEvent::KeyVerificationRequest};
         assert_eq!(event.content.transaction_id, "S0meUniqueAndOpaqueString");
         assert_eq!(event.content.from_device, "AliceDevice2");
         assert_eq!(event.content.methods, &[VerificationMethod::MSasV1]);
-        assert_eq!(event.content.timestamp, UInt::new(1559_598944869).unwrap());
+        assert_eq!(
+            event.content.timestamp,
+            UInt::new(1_559_598_944_869).unwrap()
```
Comment (jplatte) [inline src/to_device.rs]: Thanks for fixing that ugly literal :)

Comment (NerdyPepper) [reply src/to_device.rs]: :)

File: src/direct.rs (hunk 3 above)
Comment (NerdyPepper) [reply src/direct.rs]: oh i see, ill fix that up real quick

File: src/key/verification/start.rs [hunk 5]
```diff
@@ -415,10 +415,20 @@ mod tests {
             content: key_verification_start_content,
         };
 
-        assert_eq!(
-            to_string(&key_verification_start).unwrap(),
-            r#"{"content":{"from_device":"123","transaction_id":"456","method":"m.sas.v1","key_agreement_protocols":["curve25519"],"hashes":["sha256"],"message_authentication_codes":["hkdf-hmac-sha256"],"short_authentication_string":["decimal"]},"type":"m.key.verification.start"}"#
-        );
+        let json_data = json!({
+            "content":{
```
Comment (jplatte) [inline src/key/verification/start.rs]: Please add spaces after the `"key":` part here and below.

```suggestion
            "content": {
```

File: src/lib.rs [hunk 6]
```diff
@@ -111,6 +111,7 @@
 
 #![warn(rust_2018_idioms)]
 #![deny(missing_debug_implementations, missing_docs)]
+#![recursion_limit = "1024"]
```
Comment (jplatte) [inline src/lib.rs]: Please move this two lines up, so it's not mixed in between the lint attributes.

File: src/key/verification/start.rs (hunk 5 above)
Comment (NerdyPepper) [reply src/key/verification/start.rs]: alright, i have made these changes, i hope i got everything.

Comment (jplatte) [reply src/key/verification/start.rs]: If you haven't, that can still be fixed later.

**Pull request merged**